core/runtime/v2/manager.go

```diff
@@ -481,6 +512,54 @@ func (m *TaskManager) Create(ctx context.Context, taskID string, opts runtime.Cr
 	return t, nil
 }
 
+// tryStreamEvents attempts to stream shim events using bidirectional TTRPC/GRPC channel.
+func (m *TaskManager) tryStreamEvents(ctx context.Context, shim *shimTask) error {
+	// TODO: Use slices.Contains once on Go 1.21
+	supportsStreaming := false
+	for _, f := range shim.Features() {
+		if f == shimbinary.EventStreaming {
+			supportsStreaming = true
+			break
+		}
+	}
+
+	// Not supported, shim will use legacy events reporting.
+	if !supportsStreaming {
+		return nil
+	}
+
+	log.G(ctx).Info("using shim events streaming")
+
+	stream, err := shim.task.Events(context.Background(), nil)
+	if err != nil {
+		return fmt.Errorf("failed to obtain events streamer from shim: %w", err)
+	}
+
+	go func() {
+		for {
+			evt, err := stream.Recv()
+			if err != nil {
+				log.G(ctx).WithError(err).Error("failed to receive shim event from stream")
```

Comment:
If the stream was closed normally, the error was be `io.EOF`, in this case, I think we should not print it~

Reply:
Updated